### PR TITLE
Extended data for calculated properties

### PR DESCRIPTION
**Pull request by JonasDov**
imodel-native: https://github.com/iTwin/imodel-native/pull/861

### CODE REVIEW

File: docs/presentation/content/CalculatedPropertiesSpecification.md [hunk 1]
```diff
@@ -136,6 +137,23 @@ to be used in UI.
 [[include:Presentation.Content.Customization.CalculatedPropertiesSpecification.Editor.Result]]
 ```
 
+### Attribute: `extendedData`
+
+A map of [ECExpressions](./ECExpressions.md#rule-condition) whose evaluation results are used as extended data values.
```
Comment (grigasp) [inline docs/presentation/content/CalculatedPropertiesSpecification.md:142]:
```suggestion
A map of [ECExpressions](./ECExpressions.md) whose evaluation results are used as extended data values.
```

File: full-stack-tests/presentation/src/learning-snippets/content/customization/CalculatedPropertiesSpecification.test.ts [hunk 2]
```diff
@@ -159,6 +159,56 @@ describe("Learning Snippets", () => {
...
+                },
+              ],
+            },
+          ],
+        };
+        // __PUBLISH_EXTRACT_END__
+        printRuleset(ruleset);
+
+        // __PUBLISH_EXTRACT_START__ Presentation.Content.Customization.CalculatedPropertiesSpecification.ExtendedData.Result
+        // Ensure that the calculated property descriptor has `has `extendedData`property.
```
Comment (grigasp) [inline full-stack-tests/presentation/src/learning-snippets/content/customization/CalculatedPropertiesSpecification.test.ts:193]:
```suggestion
        // Ensure that the calculated property field has `extendedData` items assigned to it.
```

File: presentation/common/src/presentation-common/content/Fields.ts [hunk 3]
```diff
@@ -144,6 +145,8 @@ export class Field {
   public renderer?: RendererDescription;
   /** Property editor used to edit values of this field */
   public editor?: EditorDescription;
+  /** Extended data injected into this content item */
```
Comment (grigasp) [inline presentation/common/src/presentation-common/content/Fields.ts:148]:
```suggestion
  /** Extended data associated with this field */
```

File: presentation/common/src/presentation-common/content/Fields.ts [hunk 4]
```diff
@@ -157,6 +160,7 @@ export class Field {
    * @param priority Priority of the field
    * @param editor Property editor used to edit values of this field
    * @param renderer Property renderer used to render values of this field
+   * @param extendedData TODO
```
Comment (grigasp) [inline presentation/common/src/presentation-common/content/Fields.ts:163]:
```suggestion
   * @param extendedData Extended data associated with this field
```

File: presentation/common/src/presentation-common/rules/content/modifiers/CalculatedPropertiesSpecification.ts [hunk 5]
```diff
@@ -56,4 +56,10 @@ export interface CalculatedPropertiesSpecification {
    * Type can have these values: `string`, `boolean`, `bool`, `dateTime`, `double`, `int`, `long`
    */
   type?: `string` | `boolean` | `bool` | `dateTime` | `double` | `int` | `long`;
+
+  /**
+   * A map of [ECExpressions]($docs/presentation/customization/ECExpressions.md#rule-condition) whose
+   * evaluation results are used as extended data values.
+   */
+  extendedData?: { [key: string]: string };
```
Comment (grigasp) [inline presentation/common/src/presentation-common/rules/content/modifiers/CalculatedPropertiesSpecification.ts:64]:
```suggestion
  /**
   * A map of [ECExpressions]($docs/presentation/customization/ECExpressions.md) whose
   * evaluation results are used as extended data values.
   */
  extendedData?: { [key: string]: string };
```

File: docs/changehistory/NextVersion.md [hunk 6]
```diff
@@ -3,3 +3,10 @@ publish: false
 ---
 # NextVersion
 
+Table of contents:
+
+- [Calculated properties specification enhancements](#calculated-properties-specification-enhancements)
+
+### Calculated properties specification enhancements
```
Comment (grigasp) [inline docs/changehistory/NextVersion.md:10]: Please add a "Presentation" 2nd level section

File: docs/presentation/content/CalculatedPropertiesSpecification.md [hunk 7]
```diff
@@ -136,6 +137,23 @@ to be used in UI.
 [[include:Presentation.Content.Customization.CalculatedPropertiesSpecification.Editor.Result]]
 ```
 
+### Attribute: `extendedData`
```
Comment (grigasp) [inline docs/presentation/content/CalculatedPropertiesSpecification.md:140]: would be nice if order of attribute sections matched their order in the table

File: docs/changehistory/NextVersion.md (hunk 6 above)
Comment (JonasDov) [reply docs/changehistory/NextVersion.md]: Added

File: docs/presentation/content/CalculatedPropertiesSpecification.md (hunk 7 above)
Comment (JonasDov) [reply docs/presentation/content/CalculatedPropertiesSpecification.md]: Moved extended data attribute section to the bottom.

**Comment by aruniverse** [comment]
@mergifyio backport release/4.9.x

**Pull request merged**